app/helpers/selections_helper.rb

```diff
@@ -2,34 +2,49 @@
 require 'ostruct'
 
 module SelectionsHelper
-
   def get_forecast(selections)
-    w_api = Wunderground.new(ENV["WUNDERGROUND_API_KEY"])
+    w_api = Wunderground.new(ENV['WUNDERGROUND_API_KEY'])
 
     # user forecast is an array of forecasts
-    selections.each do |selection|
+    selections.map do |selection|
       raw_response = w_api.forecast_and_conditions_for("pws:#{selection.code}")
       response = JSON.parse(raw_response.to_json, object_class: OpenStruct)
 
-      forecast = {
-        station_id: selection.id,
-        description: selection.description,
-        observations: response.current_observation,
-        elevation: response.current_observation.observation_location.elevation
-      }
+        puts response
+        puts selection.code
+        forecast = {
+          :station_id => selection.id,
+          :description => selection.description,
+          :elevation => response.current_observation.observation_location.elevation,
+          :observations => {
+            :time => response.current_observation.observation_time_rfc822,
+            :temp => response.current_observation.temp_f,
+            :wind => response.current_observation.wind_string,
+            :precip_today => response.current_observation.precip_today_in,
```

Comment:
You should probably do `.to_f` here, not down below.
And you should do the "greater than zero" bit here too:
```
:precip_today => [response.current_observation.precip_today_in.to_f, 0].max,
: precip_1hr => [response.current_observation. precip_1hr_in.to_f, 0].max,
```

Reply:
I see, max is new to me, just read up on it